nuxeo-ai-core/src/main/java/org/nuxeo/ai/functions/SaveEnrichmentFunction.java

```diff
@@ -41,30 +42,52 @@ public class SaveEnrichmentFunction extends AbstractEnrichmentConsumer {
     public void accept(EnrichmentMetadata metadata) {
         TransactionHelper.runInTransaction(() -> CoreInstance.doPrivileged(metadata.context.repositoryName, session -> {
             DocMetadataService docMetadataService = Framework.getService(DocMetadataService.class);
+
+            log.debug("Saving enrichment for document {}.", metadata.context.documentRef);
             DocumentModel doc = docMetadataService.saveEnrichment(session, metadata);
-            if (doc != null) {
-                try {
-                    if (!doc.isImmutable()) {
-                        log.debug("Saving enrichment for document {}.", doc.getId());
-                        session.saveDocument(doc);
-                    } else {
-                        log.error("Attempt to write into an Immutable Document Model id: {}, AI Model name {}",
-                                doc.getId(), metadata.getModelName());
-                    }
-                } catch (DocumentValidationException e) {
-                    log.warn("Failed to save document enrichment data for {}; error {}", metadata.context.documentRef,
-                            e.getMessage());
-                    if (log.isDebugEnabled()) {
-                        // log field violations
-                        List<ConstraintViolation> violations = e.getReport().asList();
-                        for (ConstraintViolation violation : violations) {
-                            log.debug("Constraint: {}, Value: {}", violation.getConstraint().getDescription(),
-                                    violation.getInvalidValue());
-                        }
+
+            if (doc == null) {
+                log.warn("Failed to save enrichment for document {}.", metadata.context.documentRef);
+                return null;
+            }
+
+            log.debug("Checking if the document is checked out and if a base version exists for the document {}.",
+                    doc.getId());
+
+            DocumentRef baseVersionRef = session.getBaseVersion(doc.getRef());
+
+            if (baseVersionRef == null && !doc.isCheckedOut()) {
```

Comment:
The `!doc.isCheckedOut()` check was added in dd3b862658a35f863a8a0ab7b85739af3a58b189 (see -  https://github.com/nuxeo/nuxeo-ai/compare/14648a99f3c449402bab08e28f3c4c6c78495c30..dd3b862658a35f863a8a0ab7b85739af3a58b189 ), because there are valid cases where documents do not have base versions (such as a document that was never versioned). The check before was too strong and was causing one of our tests to fail, as shown below.

This change relates to the code path from the exception where a document is checked in and the versioning service tries to check out the document but fails as there isn't a base version:

https://github.com/nuxeo/nuxeo-lts/blob/eb3a6cda7c8a5a7d9e212e105a76bee1bb801166/modules/core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/versioning/StandardVersioningService.java#L368-L385

For a description of the `isCheckedOut()`, please see here:

https://github.com/nuxeo/nuxeo-lts/blob/eb3a6cda7c8a5a7d9e212e105a76bee1bb801166/modules/core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/DocumentModel.java#L351-L362

The test failure was the following:

```
java.lang.AssertionError: The document must have the enrichment facet
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.nuxeo.ai.enrichment.TestConfiguredStreamProcessors.testConfiguredStreamProcessor(TestConfiguredStreamProcessors.java:135)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
```

( https://github.com/nuxeo/nuxeo-ai/pull/597/checks?check_run_id=13046678534 )